src/mem/queue.rs

```diff
@@ -136,7 +136,18 @@ impl MemQueue {
         self.record_metas.is_empty()
     }
 
-    /// Returns what should be the next position.
+    /// Returns the current (last) position.
+    pub fn current_position(&self) -> Option<u64> {
```

Comment:
I think it could be written in term of `next_position().checked_sub(1)`, or alternatively, that next position be `(current_position() +1).unwrap_or(0)`, so the logic isn't duplicated